src/platform/web/ui/general/LazyListView.js

```diff
@@ -0,0 +1,251 @@
+        this._renderRange = renderRange;
+        this._renderElementsInRange();
+    }
+
+    _renderItems(items) {
+        this._childInstances = [];
+        const fragment = document.createDocumentFragment();
+        for (const item of items) {
+            const view = this._childCreator(item);
+            this._childInstances.push(view);
+            fragment.appendChild(mountView(view, this._mountArgs));
+        }
+        this._root.appendChild(fragment);
+    }
+
+    _itemsFromList(start, end) {
+        const array = [];
+        let i = 0;
+        for (const item of this._list) {
+            if (i >= start && i < end) {
+                array.push(item);
+            }
+            i = i + 1;
+        }
+        return array;
+    }
+
+    _renderElementsInRange() {
+        const { topCount, renderCount, bottomCount } = this._renderRange;
+        const paddingTop = topCount * this._itemHeight;
+        const paddingBottom = bottomCount * this._itemHeight;
+        const renderedItems = this._itemsFromList(topCount, topCount + renderCount);
+        this._root.style.paddingTop = `${paddingTop}px`;
+        this._root.style.paddingBottom = `${paddingBottom}px`;
+        for (const child of this._childInstances) {
+            child.root().remove();
+            child.unmount();
+        }
+        this._renderItems(renderedItems);
+    }
+
+    mount() {
+        const root = super.mount();
+        this._parent = el("div", {className: "LazyListParent"}, root);
+        /*
+        Hooking to scroll events can be expensive.
+        Do we need to do more (like event throttling)?
+        */
+        this._parent.addEventListener("scroll", () => this._renderIfNeeded());
+        this._initialRender();
+        return this._parent;
+    }
+
+    update(attributes) {
+        this._renderRange = null;
+        super.update(attributes);
+        this._initialRender();
+    }
+
+    loadList() {
+        if (!this._list) { return; }
+        this._subscription = this._list.subscribe(this);
+        this._childInstances = [];
+        /*
+        super.loadList() would render the entire list at this point.
+        We instead lazy render a part of the list in _renderIfNeeded
+        */
+    }
+
+
+    // If size of the list changes, re-render
+    onAdd() {
+        this._renderIfNeeded(true);
+    }
+
+    onRemove() {
+        this._renderIfNeeded(true);
+    }
+
+    onUpdate(idx, value, params) {
+        if (this._renderRange.containsIndex(idx)) {
+            const normalizedIdx = this._renderRange.normalize(idx);
+            super.onUpdate(normalizedIdx, value, params);
+        }
+    }
+    
+    recreateItem(idx, value) {
+        if (this._renderRange.containsIndex(idx)) {
+            const normalizedIdx = this._renderRange.normalize(idx);
+            super.recreateItem(normalizedIdx, value)
+        }
+    }
+
+    _renderAdditionalElement(fromIdx, toIdx) {
```

Comment:
Can we give this a name more specific to moving an item? Given it's arguments, it looks like it would only apply in that scenario.